### PR TITLE
Admin allowed node roles

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Provide the ability to restrict the modification of nodes based off roles
## Impacts

The admin GUI
## NEWS file entries

New Features
++++++++++++
- Provide the ability to Restrict the modification of node roles
## Delete branch after merge

NO

### CODE REVIEW

**Comment by rouzier** [comment]
Please Review

File: html/pfappserver/lib/pfappserver/Form/Node.pm [hunk 1]
```diff
@@ -190,6 +195,67 @@ sub validate {
...
+    my ($self) = @_;
+    my $init_object = $self->init_object;
+    return undef unless defined $init_object;
+    my $role = $self->init_object->{category};
+    return undef unless defined $role;
+    my %allowed_node_roles = map {$_ => undef} $self->_get_allowed_options('allowed_node_roles');
+    return
+        keys %allowed_node_roles == 0     ? undef
+      : exists $allowed_node_roles{$role} ? undef
+      :                                     1;
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Node.pm:256]: Seems to be a bit too much logic for the ternary operator

**Comment by julsemaan** [comment]
Seems quite big for a last minute addition.

Will require a good review + some good testing love

Would be in favor of moving it to 6.2 since 6.1 is getting quite fat

**Comment by louismunro** [comment]
Agreed. Moved to 6.2.

File: html/pfappserver/lib/pfappserver/Form/Node.pm (hunk 1 above)
Comment (rouzier) [inline html/pfappserver/lib/pfappserver/Form/Node.pm:256]: If it is not clear what is going then it make sense to change it

**Pull request merged**